miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ExperimentRestController.java

```diff
@@ -124,23 +124,26 @@
 
   @PostMapping(produces = "application/json")
   public @ResponseBody ExperimentDto create(@RequestBody ExperimentDto dto) throws IOException {
-    return RestUtils.createObject("Experiment", dto, Dtos::to, experimentService, Dtos::asDto);
+    return RestUtils.createObject("Experiment", dto, Dtos::to, experimentService,
+        expt -> Dtos.asDto(expt));
   }
 
   @PutMapping("/{experimentId}")
   public @ResponseBody ExperimentDto update(@PathVariable long experimentId, @RequestBody ExperimentDto dto) throws IOException {
-    return RestUtils.updateObject("Experiment", experimentId, dto, Dtos::to, experimentService, Dtos::asDto);
+    return RestUtils.updateObject("Experiment", experimentId, dto, Dtos::to, experimentService,
+        expt -> Dtos.asDto(expt));
   }
 
   @GetMapping(value = "/{experimentId}", produces = "application/json")
   public @ResponseBody ExperimentDto get(@PathVariable Long experimentId) throws IOException {
-    return RestUtils.getObject("Experiment", experimentId, experimentService, Dtos::asDto);
+    return RestUtils.getObject("Experiment", experimentId, experimentService,
+        expt -> Dtos.asDto(expt));
   }
 
   @GetMapping(produces = "application/json")
   public @ResponseBody List<ExperimentDto> list() throws IOException {
     Collection<Experiment> experiments = experimentService.list();
-    return experiments.stream().map(Dtos::asDto).collect(Collectors.toList());
+    return experiments.stream().map(expt -> Dtos.asDto(expt)).collect(Collectors.toList());
```

Comment:
You've made changes similar to this in a few places... If your compiler was complaining, you might want to update your Java version. I was previously using a version that had a similar bug too

Reply:
Ah, this is left over from when I had to expand it to pass in the error and warning distances. Will revert.